src/core/codestream/ojph_params.cpp

```diff
@@ -600,45 +600,45 @@ namespace ojph {
     void param_siz::read(infile_base *file)
     {
       if (file->read(&Lsiz, 2) != 2)
-        OJPH_ERROR(0x00050041, "error reading SIZ marker");
+        OJPH_ERROR(0x00050041, "error reading SIZ marker, truncated file");
       Lsiz = swap_byte(Lsiz);
       int num_comps = (Lsiz - 38) / 3;
       if (Lsiz != 38 + 3 * num_comps)
-        OJPH_ERROR(0x00050042, "error in SIZ marker length");
+        OJPH_ERROR(0x00050042, "error in SIZ marker length %d with components %d", Lsiz, num_comps);
       if (file->read(&Rsiz, 2) != 2)
-        OJPH_ERROR(0x00050043, "error reading SIZ marker");
```

Comment:
From this position downwards.
This fine granularity might be useful for a developer; it is of little use to a normal user.
The Rsiz, XSIZ, YSIZ, ... etc. are parameters within the SIZ marker segment.  See the following table.

![image](https://github.com/user-attachments/assets/a31ca3e3-006d-42c9-b4bb-9724ae0a8c93)